pkg/k8s/k8s.go

```diff
@@ -62,10 +62,12 @@ type cluster struct {
 
 // GetCluster returns a current configured cluster,
 // receives context to use, if empty uses default
-func GetCluster(context string) (Cluster, error) {
+// kubeconfig can be used to specify the config file path (overrides KUBECONFIG env)
+func GetCluster(context string, kubeconfig string) (Cluster, error) {
```

Comment:
Maybe we can use opportunity here with the option pattern , something like this ....  wdyt ? 
```
type ClusterOption func(*genericclioptions.ConfigFlags)

func WithoutKubeConfig(kubeConfig string) ClusterOption {
	return func(c *genericclioptions.ConfigFlags) {
		c.KubeConfig = &kubeConfig
	}
}

func GetCluster(opts ...ClusterOption) (Cluster, error) {
	cf := genericclioptions.NewConfigFlags(true)
	for _, opt := range opts {
		opt(cf)
	}
```

Reply:
Wasn't aware it. Thanks! Let me try using it.

Reply:
👍 agree, though I think @chen-keinan meant `WithKubeConfig` instead of `WithoutKubeConfig`, an example of this implementation is https://github.com/aquasecurity/trivy/blob/main/pkg/commands/artifact/run.go#L98-L111